File: src/main/java/com/rsoi/lab1personservice/dto/request/PersonRequest.java
package com.rsoi.lab1personservice.dto.request;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;

public class PersonRequest {
    @NotBlank
    private String name;
    private String work;
    @Min(value = 1)
    private Integer age;
    private String address;

    public PersonRequest(String name, String work, Integer age, String address) {
        this.name = name;
        this.work = work;
        this.age = age;
        this.address = address;
    }

    public PersonRequest() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getWork() {
        return work;
    }

    public void setWork(String work) {
        this.work = work;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }
}

File: DS_Lab1/Models/PersonRequest.cs
using DS_Lab1.Models;

namespace DS_Lab1.CommunicationElements
{
    public class PersonRequest
    {
        public string Name { get; set; }
        public int? Age { get; set; }
        public string? Adress { get; set; }
        public string? Work { get; set; }

        public PersonRequest(string name, int? age = null, string? adress = null, string? work = null)
        {
            Name = name;
            Age = age;
            Adress = adress;
            Work = work;
        }

        public Person ToPerson() => new(Name, Age, Adress, Work);
    }
}
